Documentation for new core 7.2 startup API (#4475)
Particular/docs.particular.net: nservicebus/dependency-injection/index_ownership_core_[7.2,).partial.md

> NServiceBus automatically registers and invokes message handlers, sagas, and other user provided extension points using a dependency injection container.

Reviewer: This line is common to both 7.1. and 7.2 so it should really be in `index.md`.

Author: Agree, I'll move

> NServiceBus supports two modes of operation, *internally managed* and *externally managed* container.

Reviewer: ```suggestion
NServiceBus supports two modes of operation for containers, *internally managed* and *externally managed*.
```

## Internally managed mode

> In *internally managed* mode, NServiceBus manages the entire lifecycle of the container from registration, component resolution to disposal.

Reviewer: ```suggestion
In *internally managed* mode, NServiceBus manages the entire lifecycle of the container including registration, component resolution, and disposal.
```

include: internallymanagedcontainer

## Externally managed mode

> In *externally managed* mode, NServiceBus registers its components in the container but does not own the container's lifecycle. The container is passed by the user via two different abstractions, one for the registration phase (`IConfigureComponents`) and one for the resolution phase (`IBuilder`).

Reviewer: ```suggestion
In *externally managed* mode, NServiceBus registers its components in the container but does not own the container's lifecycle. The container is provided by the user in two phases, one for registration (`IConfigureComponents`) and one for resolution (`IBuilder`).
```

Reviewer: @andreasohlund did you decide against this suggestion? I see you didn't commit it.

> First, the `EndpointWithExternallyManagedContainer.Create` method requires an instance of `IConfigureComponents`. At this stage the container is in the registration phase (e.g. Autofac's `ContainerBuilder`).

Reviewer: ```suggestion
During the registration phase an instance of `IConfigureComponents` is passed to the `EndpointWithExternallyManagedContainer.Create` method. For example, for Autofac's `ContainerBuilder`, this is the phase during which its type registration methods would be called.
```

snippet: ExternalPrepare

> Second, the `Start` method requires an instance of `IBuilder`. At this stage the container has already been initialized will all its registrations.

Reviewer: ```suggestion
Later, during the resolution phase, the `Start` method requires an instance of `IBuilder`. At this stage the container has already been initialized will all its registrations. For example, for Autofac's `ContainerBuilder`, this is the phase during which its `Build` method would be called.
```

snippet: ExternalStart

> NOTE: The `Adapt` methods need to be provided by the user and are container-specific. See the [ASP.NET Core sample](/samples/dependency-injection/aspnetcore/) to see how these methods are implemented based on the ASP.NET Core Dependency Injection abstractions.

Reviewer: ```suggestion
NOTE: The `Adapt` methods are implemented by the user and are container-specific. See the [ASP.NET Core sample](/samples/dependency-injection/aspnetcore/) to see how these methods are implemented based on the ASP.NET Core dependency injection abstractions.
```
